core/field.ts

```diff
@@ -1031,7 +1033,8 @@ export abstract class Field<T = any> implements IASTNodeLocationSvg,
     this.doValueUpdate_(localValue);
     if (source && eventUtils.isEnabled()) {
       eventUtils.fire(new (eventUtils.get(eventUtils.BLOCK_CHANGE))(
```

Comment:
In the case where eventOriginType == "complete_user_input", this setValue method isn't even being called. The act of closing the editor dispatches an event directly instead of calling this method. In other cases, the value is usually expected to have changed. 

However, the event being fired here still distinguishes between eventOriginType == "incomplete_user_input" and eventOriginType == undefined. In the former case, workspace change listeners can use eventOriginType to skip doing unnecessary work.